### ISSUE TITLE
Apache/Nginx/PHP-FPM not handling index.html as index file.

### ISSUE THREAD

**Issue by sarahjean** [geerlingguy/drupal-vm#617]
Working with the zen theme, which uses a static styleguide, and it looks like Apache is not loading index.html as an index file. 

To reproduce you can start up a fresh drupalvm instance, install drupal 8.2.x and the zen theme + components module. If you visit http://drupalvm.dev/themes/contrib/zen/STARTERKIT/styleguide/ you get a 404, but you can visit http://drupalvm.dev/themes/contrib/zen/STARTERKIT/styleguide/index.html directly

Have errors similar to http://serverfault.com/questions/450628/apache-2-4-php-fpm-proxypassmatch in /var/log/apache2/error.log.

**Comment by derimagia** [comment]
Took a look at this for a bit to see exactly what was happening here and why it wasn't happening in nginx. 

This answer: http://serverfault.com/a/672969 (From the link you linked) seems to have a fix for apache and the reason FilesMatch/SetHandler wasn't used before is because of the fact that drupal-vm supported Apache 2.2 before which didn't support sockets. We should take a look if we can pull this out of every virtualhost so clean everything up. We may be able to remove the DirectoryIndex at the top of the file.

Nginx didn't have this issue because in apache the path is being hit by .htaccess and then the path is being redirected to index.php and then it's failing there. Nginx has a try_files clause which checks if the file (and directory) exist which then hits the index clause...

**Comment by geerlingguy** [comment]
Taking this further, we might be able to figure out a way to use something like:

```
<FilesMatch \.php$>
   SetHandler "proxy:unix:/path/to/socket.sock|fcgi://unique-domain-name-string/"
 </FilesMatch>
```

Instead of the current `ProxyPassMatch` in a way that's slightly easier to maintain. Maybe... Another option is to try to figure out how to get ProxyPassMatch to work okay with `index.html`.

**Comment by attheshow** [comment]
Same issue here. We're switching one of our sites from being Drupal based to become a more static site with an index.html but we're attempting to run it on our VM. Right now we can view it but we have to be sure that we explicitly type "/index.html" onto the end of the domain name.

**Comment by geerlingguy** [comment]
@attheshow - Thanks for the +1! For now, the main fix is to use Nginx instead of Apache; though that's not ideal. I just haven't had the time to work on a more permanent fix for Apache.

**Comment by attheshow** [comment]
@geerlingguy No problem. So I'm new to most of this but I'm looking at the `DirectoryIndex index.html` line in the default.config.yml file (line 122) that's used as an `extra_parameter` for the dashboard. Does that do the trick I wonder? I'll try it...

**Comment by attheshow** [comment]
Nope, that didn't fix it doing the extra_parameter.

**Comment by geerlingguy** [comment]
@attheshow - It's never easy, is it :D

**Comment by attheshow** [comment]
Oooh, I just logged into the machine and edited the vhosts.conf file and commented out the ProxyPassMatch line and that appeared to fix it. I'll remove that line from my config.yml file too. This might work for others if they're doing a static site. But not for Drupal sites with static assets like what was mentioned with the Zen theme in the original report above.

**Comment by oxyc** [comment]
If someone wants this feature and uses Apache 2.4 (should be pretty much everyone), this seems to work with the minimal testing I've done.

```diff
diff --git a/default.config.yml b/default.config.yml
index ba9bcb8..0b6342f 100644
--- a/default.config.yml
+++ b/default.config.yml
@@ -106,29 +106,38 @@ apache_vhosts:
     serveralias: "www.{{ drupal_domain }}"
     documentroot: "{{ drupal_core_path }}"
     extra_parameters: |
-          ProxyPassMatch ^/(.*\.php(/.*)?)$ "fcgi://127.0.0.1:9000{{ drupal_core_path }}"
+          <FilesMatch \.php$>
+            SetHandler "proxy:unix:{{ php_fpm_listen }}|fcgi://{{ drupal_domain }}/"
+          </FilesMatch>
 
   - servername: "adminer.{{ vagrant_hostname }}"
     documentroot: "{{ adminer_install_dir }}"
     extra_parameters: |
-          ProxyPassMatch ^/(.*\.php(/.*)?)$ "fcgi://127.0.0.1:9000{{ adminer_install_dir }}"
+          <FilesMatch \.php$>
+            SetHandler "proxy:unix:{{ php_fpm_listen }}|fcgi://adminer.{{ vagrant_hostname }}/"
+          </FilesMatch>
 
   - servername: "xhprof.{{ vagrant_hostname }}"
     documentroot: "{{ php_xhprof_html_dir }}"
     extra_parameters: |
-          ProxyPassMatch ^/(.*\.php(/.*)?)$ "fcgi://127.0.0.1:9000{{ php_xhprof_html_dir }}"
+          <FilesMatch \.php$>
+            SetHandler "proxy:unix:{{ php_fpm_listen }}|fcgi://xhprof.{{ vagrant_hostname }}/"
+          </FilesMatch>
 
   - servername: "pimpmylog.{{ vagrant_hostname }}"
     documentroot: "{{ pimpmylog_install_dir }}"
     extra_parameters: |
-          ProxyPassMatch ^/(.*\.php(/.*)?)$ "fcgi://127.0.0.1:9000{{ pimpmylog_install_dir }}"
+          <FilesMatch \.php$>
+            SetHandler "proxy:unix:{{ php_fpm_listen }}|fcgi://pimpmylog.{{ vagrant_hostname }}/"
+          </FilesMatch>
 
   - servername: "{{ vagrant_ip }}"
     serveralias: "dashboard.{{ vagrant_hostname }}"
     documentroot: "{{ dashboard_install_dir }}"
     extra_parameters: |
-          ProxyPassMatch ^/(.*\.php(/.*)?)$ "fcgi://127.0.0.1:9000{{ dashboard_install_dir }}"
-          DirectoryIndex index.html
+          <FilesMatch \.php$>
+            SetHandler "proxy:unix:{{ php_fpm_listen }}|fcgi://dashboard.{{ vagrant_hostname }}/"
+          </FilesMatch>
 
 apache_remove_default_vhost: true
 apache_mods_enabled:
@@ -252,7 +261,7 @@ php_max_input_vars: "4000"
 # to instead use Apache + mod_php with an Ubuntu base box, make sure you add
 # libapache2-mod-php7.0 to `extra_packages` elsewhere in this config file.
 php_enable_php_fpm: true
-php_fpm_listen: "127.0.0.1:9000"
+php_fpm_listen: /var/run/php-fpm.sock
 
 composer_path: /usr/bin/composer
 composer_home_path: "/home/{{ drupalvm_user }}/.composer"
```

@geerlingguy when do you plan to drop support for 2.2?

**Comment by geerlingguy** [comment]
@oxyc - ASAP :)

We can ditch CentOS 6 now, I think, as that was the only OS where it was annoyingly difficult to get 2.4 and FPM working easily.

**Comment by oxyc** [comment]
Fixed in #1086.
